api4/webhook_test.go

```diff
@@ -169,6 +168,20 @@ func TestGetIncomingWebhooks(t *testing.T) {
 		t.Fatal("no hooks should be returned")
 	}
 
+	// Test Team User Case: Permission to MANAGE_INCOMING_WEBHOOKS but not to MANAGE_OTHERS_INCOMING_WEBHOOKS
+	th.AddPermissionToRole(model.PERMISSION_MANAGE_INCOMING_WEBHOOKS.Id, model.TEAM_USER_ROLE_ID)
```

Comment:
Should we remove the duplicate permission assignment / checks on line 188-191?

Reply:
I've extracted the unit test for the FilterByTeamByUser to it's own case and added the extra testing data like suggested below.